### PR TITLE
feat: Add TypeScript declaration

### LINKED ISSUE #15: Add Types
I'm considering adding typing for TypeScript which is the one I use the most,

### PR DESCRIPTION
**Pull request by Rapsssito**
Heavily inspired on https://github.com/inthepocket/react-native-wifi-reborn/blob/master/index.d.ts

Closes #15.

### CODE REVIEW

**Comment by Rapsssito** [comment]
This PR will be merged right after #43.

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{
+    "plugins": ["prettier", "@typescript-eslint"],
+    "env": {
+        "es6": true,
+        "node": true
+    },
+    "parser": "@typescript-eslint/parser",
+    "parserOptions": {
+        "ecmaVersion": 2018,
+        "sourceType": "module"
```
Comment (thibmaek) [inline .eslintrc.json]: If you want you can use your tsconfig here as the parserOption:

```json
"parserOptions": {
    "project": "./tsconfig.json"
},
```

Comment (Rapsssito) [reply .eslintrc.json]: @thibmaek, thank you! I have updated the parserOptions

**Pull request merged**